ui/components/ui/delineator/utils.ts

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {
+  BackgroundColor,
+  IconColor,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+import { AvatarIconSize, IconProps, Text } from '../../component-library';
+import { DelineatorType } from './delineator.types';
+
+const defaultIconProps = {
+  size: AvatarIconSize.Xs,
+};
+
+/*
+ * Get the icon props based on the type of delineator.
+ */
+export const getIconPropsByType = (type?: DelineatorType) => {
+  let iconProps = {};
+  const inverseIconColorProp = {
+    color: IconColor.infoInverse,
+  } as IconProps<'span'>;
+
+  switch (type) {
+    case DelineatorType.Error:
+      iconProps = {
+        iconProps: inverseIconColorProp,
+        backgroundColor: BackgroundColor.errorDefault,
+      };
+      break;
+    default:
+      iconProps = {
+        iconProps: inverseIconColorProp,
+        backgroundColor: BackgroundColor.overlayAlternative,
+      };
+  }
+  return { ...defaultIconProps, ...iconProps };
+};
```

Comment:
Is it possible to have icon without iconsProps as `inverseIconColorProp` ?

Reply:
Good call, adrdressed this in https://github.com/MetaMask/metamask-extension/pull/25610/commits/7b0a644ab351637264332f533f5e9a8c16d5ad0a